docs/user/recipes/output-csv.rst

```diff
@@ -0,0 +1,143 @@
+.. _outputting_csv_recipe:
+
+Outputting CSV Files
+====================
+
+Generating a CSV (or PDF etc) report and making it available as a downloadable
+file is a fairly common back-end service task.
+
+The easiest approach is to simply write CSV rows to an ``io.StringIO`` stream,
+and then assign its value to :attr:`resp.body <falcon.Response.body>`:
+
+.. tabs::
+
+    .. group-tab:: WSGI
+
+        .. code:: python
+
+            class Report:
+
+                def on_get(self, req, resp):
+                    output = io.StringIO()
+                    writer = csv.writer(output, quoting=csv.QUOTE_NONNUMERIC)
+                    writer.writerow(('fruit', 'quantity'))
+                    writer.writerow(('apples', 13))
+                    writer.writerow(('oranges', 37))
+
+                    resp.content_type = 'text/csv'
+                    resp.downloadable_as = 'report.csv'
+                    resp.body = output.getvalue()
+
+    .. group-tab:: ASGI
+
+        .. code:: python
+
+            class Report:
+
+                async def on_get(self, req, resp):
+                    output = io.StringIO()
+                    writer = csv.writer(output, quoting=csv.QUOTE_NONNUMERIC)
+                    writer.writerow(('fruit', 'quantity'))
+                    writer.writerow(('apples', 13))
+                    writer.writerow(('oranges', 37))
+
+                    resp.content_type = 'text/csv'
+                    resp.downloadable_as = 'report.csv'
+                    resp.body = output.getvalue()
+
+Here we are setting the response ``Content-Type`` to ``"text/csv"`` as
+recommended by `RFC 4180 <https://tools.ietf.org/html/rfc4180>`_, and assigning
+the downloadable file name ``report.csv`` via the ``Content-Disposition``
+header (see also: :ref:`serve-downloadable-as`).
+
+Streaming Large CSV Files on the Fly
+------------------------------------
+
+If generated CSV responses are expected to be very large, it might be worth
```

Comment:
"If generated CSV" -> "If the generated CSV" ?

Reply:
Umm, I articles ain't my forte, but I was referring to _generated CSV responses_ in plural, I don't think the definite article is needed here.
Maybe we could change to "_If the generated CSV is expected to..._"?